src/fluree/db/query/fql/parse.cljc

```diff
@@ -575,22 +575,22 @@
                 triples
                 (map vector (range) list))
 
+        ;; literal object
         (some? value)
-        (let [obj-cmp (if (v/variable? value)
+        (let [m*      (cond-> m language (assoc :lang language))
+              obj-cmp (if (v/variable? value)
                         (parse-variable value)
-                        (cond-> {::where/val value}
-                          (or m language) (assoc ::where/m (cond-> m language (assoc :lang language)))
-                          type (assoc ::where/datatype type)))]
+                        (where/anonymous-value value type m*))]
           (conj triples [subj-cmp pred-cmp obj-cmp]))
 
+        ;; ref object
         :else
-        (let [ref-cmp (if (nil? id)
-                        {::where/val (temp-bnode-id bnode-counter) ::where/datatype const/iri-id}
-                        (cond-> {::where/val id ::where/datatype const/iri-id}
-                          m (assoc ::where/m m)))
+        (let [ref-cmp (-> (where/match-iri (if (nil? id) (temp-bnode-id bnode-counter) id))
+                          (assoc ::where/datatype const/iri-id)
+                          (cond-> m (assoc ::where/meta m)))
               v-map* (if (nil? id)
                        ;; project newly created bnode-id into v-map
-                       (assoc v-map :id (::where/val ref-cmp))
+                       (assoc v-map :id (::where/iri ref-cmp))
```

Comment:
Any reason why you didn't use the `get-iri` function you just added? No rush to change it. There are a ton of direct `::where/get-iri` calls sprinkled throughout the code that should probably wait for another refactor.

Reply:
Nope, that was an oversight, fixed.